src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_TabKey.cs

```diff
@@ -67,60 +64,91 @@ private bool TryInvokeSnippetCompletion(TabKeyCommandArgs args, CompletionServic
 
             var text = subjectBuffer.AsTextContainer().CurrentText;
 
-            // Delete the ? and invoke completion
+            // If the user types "<line start><spaces><question><tab>"
```

Comment:
I find the comment slightly confusing. The editor should also shows the full snippet picker UI in the case of `<whatever><identifier><spaces><question><tab>`. It only has to do with what's _immediately_ to the left of the question mark (space or identifier), right? `QuestionMarkIsPrecededByIdentifierAndWhitespace` seems weirdly complicated for this task, unless I'm missing some nuance.